zookeeper-api/src/test/java/org/apache/helix/zookeeper/zkclient/TestZkClientPersistWatcher.java

```diff
@@ -108,6 +110,47 @@ public void handleChildChange(String parentPath, List<String> currentChilds)
     zkClient.close();
   }
 
+  @Test(dependsOnMethods = "testZkClientChildChange")
+  void testZkClientPersistRecursiveChange() throws Exception {
```

Comment:
May I suggest we have test coverage on: 
1. persistent watch register and removal
2. Concurrent tests with multiple watchers on same path

Reply:
TFTR！ Yes this was the plan. Right now it is a dummy test since no impl is add in zkMetaClient right now.